Fixes and documentation updates for Release 0.9

pjaaskel opened the pull request:
With these, the #142 status goes to:

```
97% tests passed, 23 tests failed out of 671

Label Time Summary:
cuda        =  63.34 sec*proc (27 tests)
internal    =  84.68 sec*proc (80 tests)

Total Test time (real) = 665.05 sec

The following tests did not run:
	641 - sycl_chip_interop (Skipped)
	642 - sycl_chip_interop_usm (Skipped)

The following tests FAILED:
	100 - Unit_hipMemcpy_HalfMemCopy (Failed)
	127 - Unit_hipMemset_InvalidPtrTests (Failed)
	205 - Unit_hipMemcpyAsync_KernelLaunch - int (Failed)
	206 - Unit_hipMemcpyAsync_KernelLaunch - float (Failed)
	207 - Unit_hipMemcpyAsync_KernelLaunch - double (Failed)
	500 - Unit_hipStreamGetFlags_BasicFunctionalities (Failed)
	508 - Unit_hipMultiStream_sameDevice (SEGFAULT)
	519 - Unit_hipStreamCreate_MultistreamBasicFunctionalities (Failed)
	545 - Unit_hipDeviceSynchronize_Functional (Failed)
	562 - Unit_hipStreamPerThread_Basic (Failed)
	571 - Unit_hipStreamPerThread_MultiThread (SEGFAULT)
	572 - Unit_hipStreamPerThread_DeviceReset_1 (SEGFAULT)
	579 - Stress_hipMalloc (Failed)
	581 - Stress_hipMemcpy_multiDevice-AllAPIs - int (Timeout)
	582 - Stress_hipMemcpy_multiDevice-AllAPIs - size_t (Timeout)
	583 - Stress_hipMemcpy_multiDevice-AllAPIs - long double (Failed)
	585 - activemask (Failed)
	592 - fp16 (Subprocess aborted)
	601 - 2d_shuffle (Failed)
	609 - hipStreamSemantics (Failed)
	613 - hipConstantTestDeviceSymbol (Subprocess aborted)
	639 - hipAddressingModes (Failed)
	662 - cuda-blackscholes (Failed)
Errors while running CTest
```
With this race condition workaround patch for #152 

```C++
--- a/src/CHIPBindings.cc
+++ b/src/CHIPBindings.cc
@@ -2096,6 +2096,9 @@ hipError_t hipMemcpy(void *Dst, const void *Src, size_t SizeBytes,
   NULLCHECK(Dst);
   CHECK(Src);

+  // See https://github.com/CHIP-SPV/chip-spv/issues/152.
+  // Let's be safe first, and optimize the asynch behavior later.
+  hipDeviceSynchronize();
   if (SizeBytes == 0)
     RETURN(hipSuccess);
```

it goes down to
```
97% tests passed, 18 tests failed out of 671

Label Time Summary:
cuda        =  62.26 sec*proc (27 tests)
internal    =  83.08 sec*proc (80 tests)

Total Test time (real) = 392.48 sec

The following tests did not run:
	641 - sycl_chip_interop (Skipped)
	642 - sycl_chip_interop_usm (Skipped)

The following tests FAILED:
	127 - Unit_hipMemset_InvalidPtrTests (Failed)
	173 - Unit_hipMemcpyWithStream_MultiThread (SEGFAULT)
	205 - Unit_hipMemcpyAsync_KernelLaunch - int (Failed)
	206 - Unit_hipMemcpyAsync_KernelLaunch - float (Failed)
	207 - Unit_hipMemcpyAsync_KernelLaunch - double (Failed)
	500 - Unit_hipStreamGetFlags_BasicFunctionalities (Failed)
	508 - Unit_hipMultiStream_sameDevice (SEGFAULT)
	519 - Unit_hipStreamCreate_MultistreamBasicFunctionalities (Failed)
	545 - Unit_hipDeviceSynchronize_Functional (Failed)
	562 - Unit_hipStreamPerThread_Basic (Failed)
	571 - Unit_hipStreamPerThread_MultiThread (Subprocess aborted)
	572 - Unit_hipStreamPerThread_DeviceReset_1 (Subprocess aborted)
	579 - Stress_hipMalloc (Failed)
	601 - 2d_shuffle (Failed)
	609 - hipStreamSemantics (Failed)
	613 - hipConstantTestDeviceSymbol (Subprocess aborted)
	639 - hipAddressingModes (Failed)
	662 - cuda-blackscholes (Failed)
```
I didn't test with LZ, so someone with a functioning LZ config please test, review and pull in. Thanks!

pjaaskel commented:
There's an accidental submodule change that got in, removing...

pjaaskel commented:
Sigh.. did it again. Still not happy with submodules :)

pvelesko commented on CMakeLists.txt:
```diff
@@ -32,6 +32,10 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-duplicate-decl-specifier \
 -Wno-delete-abstract-non-virtual-dtor -Wno-deprecated-declarations -Wunused-command-line-argument -gdwarf-4")
 # end temporary
 
+# Currently we require symlinks to clang and clang++ to be found in PATH.
+set(CMAKE_C_COMPILER clang CACHE STRING "Clang C compiler" FORCE)
+set(CMAKE_CXX_COMPILER clang++ CACHE STRING "Clang C++ compiler" FORCE)
+
```
What if someone does want to use `clang++-14` explicitly?

pjaaskel replied:
Yes let's merge the HIPCC integration if it fixes this annoying user-facing problem. I'll drop this patch from the PR.

pvelesko replied:
@pjaaskel HIPCC probably needs some additional testing,fyi

pvelesko replied:
I don't think it's intrusive - it worked for everything I tested with but I have the symlinks. Either way, this is something that can go into 1.0 - I'm mostly concerned with unit test pass rate now.

pjaaskel replied:
@pvelesko  ^this. I reverted it, let's think about it in 1.0?

pvelesko replied:
ok.

pjaaskel commented:
@pjaaskel I removed that CXX commit from this PR and added your OpenCL fix. If it doesn't regress for your test envs, can you push it to Release-0.9, thanks!

pvelesko commented:
> CXX commit

What CXX commit?

pvelesko commented:
2 more commits in OpenCLFixes @pjaaskel please cherry pick into whatever branch is appropriate

pjaaskel commented:
Can I push this in Release-0.9 first? Does it look OK?

pvelesko commented:
LGTM and I will cherry pick into Release-0.9 myself @pjaaskel
